aptos-move/framework/src/natives/code.rs

```diff
@@ -36,10 +37,15 @@ pub struct PackageMetadata {
     pub name: String,
     /// The upgrade policy of this package.
     pub upgrade_policy: UpgradePolicy,
+    /// Build info, in BuildInfo.yaml format
+    pub build_info: String,
     /// The package manifest, in the Move.toml format.
     pub manifest: String,
     /// The list of modules installed by this package.
     pub modules: Vec<ModuleMetadata>,
+    /// Error map, in internal encoding
+    #[serde(with = "serde_bytes")]
+    pub error_map: Vec<u8>,
```

Comment:
From a usability perspective, it might be useful to actually have a vec<u8> to type converter, so that folks don't have to figure this out by digging around too much. The entire path here is a bit ambiguous for error_map, source_map, and abi.

Reply:
I'm not sure about this. This pattern is actually common practice in protobuf and similar. You don't deserialize more than the basic use cases. Better would be indeed, you fetch only the data you need...

Perhaps you mean we introduce a wrapper to represent serialized data, so its easier identifiable of what type it is? Like

```
struct SerializedData { type: String, blob: Vec<u8> }
```

This would then also appear on Move level. In JSON, It puts the entire thing in an object and documents what its supposed to be. The interpretation of the type and blob can be fully in the SDK, so there are no version issues for this data on server side (another reason for this design pattern of embedded serialized data).